transcode/transcode_jobs.go

```diff
@@ -21,16 +21,21 @@ type ParallelTranscoding struct {
 }
 
 func NewParallelTranscoding(sourceSegmentURLs []clients.SourceSegment, work func(segment segmentInfo) error) *ParallelTranscoding {
+	totalSegs := len(sourceSegmentURLs)
 	jobs := &ParallelTranscoding{
-		queue:         make(chan segmentInfo, len(sourceSegmentURLs)),
+		queue:         make(chan segmentInfo, totalSegs),
 		errors:        make(chan error, 100),
 		work:          work,
 		isRunning:     true,
-		totalSegments: len(sourceSegmentURLs),
+		totalSegments: totalSegs,
 	}
 	// post all jobs on buffered queue for goroutines to process
 	for segmentIndex, u := range sourceSegmentURLs {
-		jobs.queue <- segmentInfo{Input: u, Index: segmentIndex}
+		if segmentIndex == totalSegs-1 {
+			jobs.queue <- segmentInfo{Input: u, Index: segmentIndex, IsLastSegment: true}
+		} else {
+			jobs.queue <- segmentInfo{Input: u, Index: segmentIndex, IsLastSegment: false}
+		}
```

Comment:
Might be nicer to do
```suggestion
		jobs.queue <- segmentInfo{Input: u, Index: segmentIndex, IsLastSegment: segmentIndex == totalSegs-1}
```